internal/file/file.go

```diff
@@ -13,9 +13,96 @@
 
 package file
 
-import "regexp"
+import (
+	"errors"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+// ErrNotRegularFile is returned when the file is not an regular file.
+var ErrNotRegularFile = errors.New("not regular file")
+
+// ErrNotDirectory is returned when the path is not a directory.
+var ErrNotDirectory = errors.New("not directory")
 
 // IsValidFileName checks if a file name is cross-platform compatible
 func IsValidFileName(fileName string) bool {
 	return regexp.MustCompile(`^[a-zA-Z0-9_.-]+$`).MatchString(fileName)
 }
+
+// CopyToDir copies the src file to dst dir. Existing file will be overwritten.
+// src file permission is preserved.
+func CopyToDir(src, dst string) error {
+	sourceFileStat, err := os.Stat(src)
+	if err != nil {
+		return err
+	}
+	if !sourceFileStat.Mode().IsRegular() {
+		return ErrNotRegularFile
+	}
+	source, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+	defer source.Close()
+	if err := os.MkdirAll(dst, 0777); err != nil {
+		return err
+	}
```

Comment:
`777` (all can read/write/search) in most of the scenario is bad permission. This means any user of the system can disrupt signing and verification operation.
If we want plugin to be sharable with other users we can update permission to read/search (No write)

Reply:
@priteshbandi I agree. Changing the directory permission to `0755`. This means that:
For `group` and `other`, they can only `list files under the dir` and `enter the dir and access the files inside`. However, they cannot `create or delete` files in the dir.